### PR TITLE
contrib/gptfdisk: new package (1.0.9)

### PR DESCRIPTION
**Pull request by reocat**

### CODE REVIEW

File: main/gdisk/patches/makefile.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- Makefile	2023-10-21 00:13:47.131679080 +0300
++++ Makefile	2023-10-21 00:46:48.279265063 +0300
+@@ -162,6 +162,8 @@
+ lint:	#no pre-reqs
+ 	lint $(SRCS)
+ 
++check: 	#empty 
++
```
Comment (nekopsykose) [inline main/gdisk/patches/makefile.patch:8]: if there's no check just do `options = ["!check"]` and you don't need this

File: main/gdisk/template.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+pkgname = "gdisk"
```
Comment (nekopsykose) [inline main/gdisk/template.py:1]: the actual project name is `gptfdisk`.

**Comment by nekopsykose** [comment]
you most likely want the 00 patches from https://git.alpinelinux.org/aports/tree/main/gptfdisk?id=84a227baf001b6e0208e3352b294e4d7a40e93de too since they fix crashes with newer libraries here

File: main/gdisk/patches/makefile.patch (hunk 1 above)
Comment (nekopsykose) [reply main/gdisk/patches/makefile.patch:8]: also it seems like the test is in `builddir/gdisk_test.sh`, so you can run that from `self.do` when overriding `do_check`.

Comment (nekopsykose) [reply main/gdisk/patches/makefile.patch:8]: ah, there is a target for it. you can do `make_check_target = "test"`

Comment (nekopsykose) [reply main/gdisk/patches/makefile.patch:8]: (but that runs it with bash, when you can manually call `sh` on it and it probably works, if you want to skip the dep)

Comment (reocat) [reply main/gdisk/patches/makefile.patch]: > if there's no check just do `options = ["!check"]` and you don't need this

ERROR: lint failed: check disabled but no reason given

**Comment by q66** [comment]
please keep your changes squashed to one commit per template

**Comment by q66** [comment]
use interactive rebase to squash all the commits together

also format the template with `black` and make sure it passes `flake8`

**Comment by q66** [comment]
perhaps you want to read https://github.com/chimera-linux/cports/blob/master/USING_GIT.md

**Comment by q66** [comment]
also, this package should go to contrib/, not main/

**Comment by reocat** [comment]
> also, this package should go to contrib/, not main/

done too

**Comment by q66** [comment]
CI fails

**Comment by reocat** [comment]
> CI fails

Check is disabled now, should work better

File: contrib/gptfdisk/template.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "ncurses-devel",
+    "popt-devel",
+]
+pkgdesc = "Fdisk-like partitioning tool for GPT disks"
+maintainer = "reocat <ng.ct_ml@tuta.io>"
+license = "GPL-2.0-or-later"
+url = "https://sourceforge.net/projects/gptfdisk"
+source = f"$(SOURCEFORGE_SITE)/{pkgname}/{pkgver}/{pkgname}-{pkgver}.tar.gz"
+sha256 = "dafead2693faeb8e8b97832b23407f6ed5b3219bc1784f482dd855774e2d50c2"
+# no check provided
```
Comment (q66) [inline contrib/gptfdisk/template.py:20]: this is a lazy way out

cosnidering gptfdisk provides a test suite and it's capable of mostly passing, you should patch out the part that is not relevant for the test environment and let the rest run

Comment (reocat) [reply contrib/gptfdisk/template.py]: Should be better now

Comment (reocat) [reply contrib/gptfdisk/template.py]: But I can't check it in GitHub CI

**Comment by reocat** [comment]
Ah, forgot to remove doas while tested some stuff, should be better now

**Comment by q66** [comment]
I looked into the failing tests here and it seems they're just flaky (i guess it's missing some syncs) so it's fine to keep the offending ones patched out

File: contrib/gptfdisk/template.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+pkgname = "gptfdisk"
+pkgver = "1.0.9"
+pkgrel = 0
+build_style = "makefile"
+make_cmd = "gmake"
+make_dir = "."
+make_check_target = "test"
+hostmakedepends = [
```
Comment (q66) [inline contrib/gptfdisk/template.py:8]: all of these except `gmake` should be `makedepends`

File: contrib/gptfdisk/template.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "linux-headers",
+    "libuuid-devel",
+    "ncurses-devel",
+    "popt-devel",
+]
+pkgdesc = "Fdisk-like partitioning tool for GPT disks"
+maintainer = "reocat <ng.ct_ml@tuta.io>"
+license = "GPL-2.0-or-later"
+url = "https://sourceforge.net/projects/gptfdisk"
+source = f"$(SOURCEFORGE_SITE)/{pkgname}/{pkgver}/{pkgname}-{pkgver}.tar.gz"
```
Comment (q66) [inline contrib/gptfdisk/template.py:19]: you don't have to substitute pkgname here

**Comment by q66** [comment]
fix the remaining two comments i raised, and then it's lgtm

File: contrib/gptfdisk/template.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+sha256 = "dafead2693faeb8e8b97832b23407f6ed5b3219bc1784f482dd855774e2d50c2"
```
Comment (q66) [inline contrib/gptfdisk/template.py:20]: also add `hardening = ["vis", "cfi"]`

**Pull request merged**